vllm/model_executor/parallel_utils/pynccl.py

```diff
@@ -219,25 +220,22 @@ def __init__(
                                     store=store,
                                     group_name=group_name,
                                     pg_options=pg_options)
-        self.world_size = dist.get_world_size()
-        self.rank = dist.get_rank()
-        torch.cuda.set_device(self.rank)
-        if self.rank == 0:
+        torch.cuda.set_device(local_rank)
+        if rank == 0:
             self.unique_id = ncclGetUniqueId()
         else:
             self.unique_id = NcclUniqueId()
-        tensor = torch.ByteTensor(list(self.unique_id.internal)).cuda(
-            self.rank)
+        tensor = torch.ByteTensor(list(
+            self.unique_id.internal)).cuda(local_rank)
         dist.broadcast(tensor, src=0)
         byte_list = tensor.cpu().tolist()
-        self.unique_id = NcclUniqueId()
```

Comment:
This line seems useless, as `self.unique_id` has already defined in line 227.

Reply:
Seems good to me.